src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -985,71 +1093,49 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
+     * combinations of matching trip and service IDs.
      */
-    private boolean containsField(Collection<Field> fields, String fieldName) {
-        for (Field field : fields) if (field.name.equals(fieldName)) return true;
-        return false;
-    }
-
-    /** Checks that any of a set of errors is of the type {@link NewGTFSErrorType#DUPLICATE_ID}. */
-    private boolean hasDuplicateError(Set<NewGTFSError> errors) {
-        for (NewGTFSError error : errors) {
-            if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) return true;
+    private MergeStrategy getMergeStrategy(List<FeedToMerge> feedsToMerge) throws IOException {
+        // Iterate over both feeds to collect all trip and service IDs.
+        for (int i = 0; i < feedsToMerge.size(); i++) {
+            FeedToMerge feed = feedsToMerge.get(i);
+            Set<Table> tablesToCheck = Sets.newHashSet(Table.TRIPS, Table.CALENDAR, Table.CALENDAR_DATES);
+            for (Table table : tablesToCheck) {
+                feed.idsForTable.get(table).addAll(getIdsForTable(feed.zipFile, table));
+            }
         }
-        return false;
-    }
-
-    /** Get table-scoped value used for key when remapping references for a particular feed. */
-    private static String getTableScopedValue(Table table, String prefix, String id) {
-        return String.join(":",
-            table.name,
-            prefix,
-            id);
-    }
-
-    /**
-     * Helper class that collects the feed version and its zip file. Note: this class helps with sorting versions to
-     * merge in a list collection.
-     */
-    private class FeedToMerge {
-        public FeedVersion version;
-        public ZipFile zipFile;
-
-        FeedToMerge(FeedVersion version) throws IOException {
-            this.version = version;
-            this.zipFile = new ZipFile(version.retrieveGtfsFile());
+        FeedToMerge futureFeed = feedsToMerge.get(0);
+        FeedToMerge activeFeed = feedsToMerge.get(1);
+        Set<String> activeTripIds = activeFeed.idsForTable.get(Table.TRIPS);
+        Set<String> futureTripIds = futureFeed.idsForTable.get(Table.TRIPS);
+        Set<String> activeServiceIds = new HashSet<>();
+        activeServiceIds.addAll(activeFeed.idsForTable.get(Table.CALENDAR));
+        activeServiceIds.addAll(activeFeed.idsForTable.get(Table.CALENDAR_DATES));
+        Set<String> futureServiceIds = new HashSet<>();
+        futureServiceIds.addAll(futureFeed.idsForTable.get(Table.CALENDAR));
+        futureServiceIds.addAll(futureFeed.idsForTable.get(Table.CALENDAR_DATES));
+        boolean serviceIdsMatch = activeServiceIds.equals(futureServiceIds);
+        intersectingTripIds = Sets.intersection(activeTripIds, futureTripIds);
+        tripsOnlyInActiveFeed = Sets.difference(activeTripIds, futureTripIds);
+        tripsOnlyInFutureFeed = Sets.difference(futureTripIds, activeTripIds);
```

Comment:
All these above intermediate variables make it harder to follow the logic.

Reply:
Do you have a suggestion for improved clarity?

Reply:
Maybe move that code inside class `FeedToMerge`?

Reply:
See 1cfd32e5362ab68b6d7b440379db98110c80299a